src/main/java/de/tum/in/www1/artemis/service/connectors/gitlabci/GitLabCIService.java

```diff
@@ -0,0 +1,222 @@
+package de.tum.in.www1.artemis.service.connectors.gitlabci;
+
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.gitlab4j.api.GitLabApi;
+import org.gitlab4j.api.GitLabApiException;
+import org.gitlab4j.api.models.Project;
+import org.gitlab4j.api.models.Trigger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.enumeration.RepositoryType;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.exception.ContinuousIntegrationException;
+import de.tum.in.www1.artemis.exception.GitLabCIException;
+import de.tum.in.www1.artemis.repository.FeedbackRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingSubmissionRepository;
+import de.tum.in.www1.artemis.service.BuildLogEntryService;
+import de.tum.in.www1.artemis.service.UrlService;
+import de.tum.in.www1.artemis.service.connectors.AbstractContinuousIntegrationService;
+import de.tum.in.www1.artemis.service.connectors.CIPermission;
+import de.tum.in.www1.artemis.service.connectors.ConnectorHealth;
+import de.tum.in.www1.artemis.service.dto.AbstractBuildResultNotificationDTO;
+
+@Profile("gitlabci")
+@Service
+public class GitLabCIService extends AbstractContinuousIntegrationService {
+
+    private static final Logger log = LoggerFactory.getLogger(GitLabCIService.class);
+
+    // TODO: Generate an URL and make it accessible via the API
+    public static final String CI_CONFIG_URL = "https://home.in.tum.de/~scbe/Artemis/gitlabci-runners/getting-started-template.yml";
```

Comment:
Can you add this file to the Artemis souces? Or does it need to be on an external URL?

Reply:
In this context, it must be an external URL from which the CI configuration is fetched (https://docs.gitlab.com/ee/ci/pipelines/settings.html#specify-a-custom-cicd-configuration-file). Therefore, I have added this as a temporary solution until I have implemented the corresponding API endpoint in Artemis.
This file is the official [getting started template](https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/gitlab/ci/templates/Getting-Started.gitlab-ci.yml). I could just link to the raw file here if you think that's better.

Reply:
What exactly is the purpose of this file? Do we need one configuration for everything? Or would it make sense to have a different configuration for every exercise?

Reply:
We plan to have a different configuration for each exercise. This is just the temporary solution to check if the setup of GitLab CI works. In the following PRs, I will implement the API endpoint and remove the link. So this link is not really meant to be for production (that's also why we do not merge in develop).

Reply:
I'm ok with this being a temporary solution, but we should not keep this long.